pinot-common/src/test/java/org/apache/pinot/common/utils/config/TableConfigSerDeTest.java

```diff
@@ -64,7 +64,13 @@
 
 
 public class TableConfigSerDeTest {
-
+  @Test
+  public void testTableConfigBuilderMetricsColumnOptimizationOptions() {
+    TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName("testTable").
+        setOptimizeDictionaryForMetrics(true).setNoDictionarySizeRatioThreshold(0.72).build();
+    assertEquals(tableConfig.getIndexingConfig().getNoDictionarySizeRatioThreshold(), 0.72);
+    assertEquals(tableConfig.getIndexingConfig().isOptimizeDictionaryForMetrics(), true);
```

Comment:
can we add the ser/de test as well?

From other test:
```
      // Serialize then de-serialize
      TableConfig tableConfigToCompare = JsonUtils.stringToObject(tableConfig.toJsonString(), TableConfig.class);
      assertEquals(tableConfigToCompare, tableConfig);
      checkQuotaConfig(tableConfigToCompare);

      tableConfigToCompare = TableConfigUtils.fromZNRecord(TableConfigUtils.toZNRecord(tableConfig));
      assertEquals(tableConfigToCompare, tableConfig);
      checkQuotaConfig(tableConfigToCompare);
```
```

Reply:
Added.